### ISSUE TITLE
Seeking Guidance on Reproduction – Observed Gap with Reported Results

### ISSUE THREAD

**Issue by SiyunZhao** [RUCAIBox/R1-Searcher#22]
Thank you very much for your excellent work and for sharing the code.

I have been working on reproducing the test results using the provided implementation. However, I observed a noticeable gap between the results I obtained and those reported in the paper. It’s possible that I may have missed some important details during the process.

To clarify, I did not retrain the model; I only performed the evaluation. I followed the encoding and FAISS index building steps in the Quick Start guide, and used the pretrained model `XXsongLALA/Qwen-2.5-7B-base-RAG-RL` available on HuggingFace to generate answers. I also used the Quick Start scripts for evaluation and metric calculation. On the HotpotQA Qwen version test set, my result for Cover-EM was **52.4%**, which is noticeably lower than the **65.4%** reported in the paper.

<img width="475" alt="Image" src="https://github.com/user-attachments/assets/60aacaf4-ee3d-4161-b0c0-b04d0721eccf" />

Here's a brief overview of my procedure:

- I used the dataset and preprocessing method shared in [Issue #11] https://rocketqa.bj.bcebos.com/corpus/nq.tar.gz. I matched titles in `para.title.txt` with chunks in `para.txt`, and concatenated them using the format:  
  `f'Title: {title} Text: {para}'`
  Here is how the data looks like:

![Image](https://github.com/user-attachments/assets/0d8ba0dd-89c5-4935-9ade-78f385e545a9)

- ran `build_corpus_embedding.py` to generate the embeddings, and `build_corpus_index.py` to build the FAISS index.
- pointed `wiki_corpus_load.py` to the above files to start the retrieval service:
  ```
  python train/wiki_corpus_load.py hotpotqa 5004 &
  ```
- generated answers using:
  ```
  python evaluation/eval_search_loacl.py --gpu_id 0 --temp 0.0 --port 5004 --prompt_type v0 --src_file data/eval_set/hotpotqa_500.jsonl --model_path XXsongLALA/Qwen-2.5-7B-base-RAG-RL
  ```
- calculated the metrics using:
  ```
  python evaluation/metric_calc_rule.py /home/aiscuser/R1-Searcher/data/eval_set/hotpotqa_500-XXsongLALAQwen-2.5-7B-base-RAG-RL_base_temp0.0_typev0.jsonl
  ```

I would sincerely appreciate any advice or suggestions you could kindly offer to help me understand this performance gap and better align my reproduction with the reported results.

Thank you again for your time and support.

**Comment by ignorejjj** [comment]
@SiyunZhao We replicated R1 Searcher based on the FlashRAG framework with the trained qwen model provided by the author and obtained results similar to yours, with a Cover EM value of 51.9 on Hotpotqa. The detail results are as follows:

![Image](https://github.com/user-attachments/assets/721eae8b-a59c-4e0d-92b2-c5896a58049a)

We fully follow the author's process, including prompt settings and document formatting. I think the inconsistency with the results in the author's paper may be due to the following reasons:
1. We rely solely on identifying the content between<answer>and</answer>as the answer, and using the complete output may result in higher accuracy.
2. We are using Wikipedia 2018 provided by Flashrag, and the segmentation method is similar to the killer used by the author, but the search results may have slight differences.
3.  We use e5-base-v2 as retriever, instead of bge-large-en, maybe use bge-large can achieve better results?

But I think these three points are difficult to bring about a ten point improvement, and I wonder if the author has any tricks in the evaluation.

Detail guidence can be found at the FlashRAG document[https://github.com/RUC-NLPIR/FlashRAG/blob/main/docs/original_docs/reproduce_experiment.md]. I hope it can make it easier for those who want to reproduce.

**Comment by SiyunZhao** [comment]
Hi @SsmallSong , thanks for your reply!

Could you give some scripts on **how to extract abs** from the corpus link you provided? As the commands on Quick Starts imply the copus come from https://rocketqa.bj.bcebos.com/corpus/nq.tar.gz :

![Image](https://github.com/user-attachments/assets/1e8758fa-e740-4d8d-af5b-d8bd6567f314)

And after downloading and uncompress the data, it is **not so clear for us on how to extract "abs"** from the full list of chunked data. The origin data structure is as follows:

![Image](https://github.com/user-attachments/assets/9b690e53-ecf0-4de5-8969-d971c95331ec)

If the extraction of abstracts is a new and important task scenario, could you please publish these data to huggingface? Your scripts on processing the nq data to title-abs chunks will also help a lot.

**Comment by Engineer-D** [comment]
Hi @SiyunZhao 

I noticed you were able to reproduce the results that @smallsong got. I tried replicating the process using the data in samples, but my evaluation scores were much lower (an EM of 8, whereas you reached 52.4). Would you mind sharing your source code or dataset so I can understand the differences and hopefully achieve better accurate results?

Thanks!

**Comment by SsmallSong** [comment]
> Hi [@SsmallSong](https://github.com/SsmallSong) , thanks for your reply!
> 
> Could you give some scripts on **how to extract abs** from the corpus link you provided? As the commands on Quick Starts imply the copus come from https://rocketqa.bj.bcebos.com/corpus/nq.tar.gz :
> 
> ![Image](https://github.com/user-attachments/assets/1e8758fa-e740-4d8d-af5b-d8bd6567f314)
> 
> And after downloading and uncompress the data, it is **not so clear for us on how to extract "abs"** from the full list of chunked data. The origin data structure is as follows:
> 
> ![Image](https://github.com/user-attachments/assets/9b690e53-ecf0-4de5-8969-d971c95331ec)
> 
> If the extraction of abstracts is a new and important task scenario, could you please publish these data to huggingface? Your scripts on processing the nq data to title-abs chunks will also help a lot.

OK. In the subsequent versions, I will upload the processed training knowledge base to Hugging Face.
